lib/commons/dom/create-grid.js

```diff
@@ -255,26 +255,41 @@ function getStackingOrder(vNode, parentVNode) {
     vNode.getComputedStylePropertyValue('position') !== 'static';
   const floated = vNode.getComputedStylePropertyValue('float') !== 'none';
 
-  // flex and grid items can use z-index even if position: static
-  // @see https://www.w3.org/TR/css-flexbox-1/#painting
-  // @see https://www.w3.org/TR/css-grid-1/#z-order
-  if (
-    !['auto', '0'].includes(zIndex) &&
-    (positioned || isFlexOrGridContainer(parentVNode))
-  ) {
-    // if a positioned element has a z-index > 0, find the first
+  if (isStackingContext(vNode, parentVNode)) {
+    // if an element creates a stacking context, find the first
     // true stack (not a "fake" stack created from positioned or
     // floated elements without a z-index) and create a new stack at
     // that point (step #5 and step #8)
     // @see https://www.w3.org/Style/css2-updates/css2/zindex.html
-    while (stackingOrder.find(value => value % 1 !== 0)) {
-      const index = stackingOrder.findIndex(value => value % 1 !== 0);
-      stackingOrder.splice(index, 1);
+    const index = stackingOrder.findIndex(
+      value => (value < 1 && value > 0.2) || value === 0
```

Comment:
These magic numbers are difficult to understand without studying the code. I switched them over to constants in my refactor PR. Please have a look